### PR TITLE
feat: added demo files from elasticsearch repo to opensearch repo

### PR DESCRIPTION
**Pull request by JohannesDaniel**
### Description
The demo facilitates the understanding of LTR and is also suitable for a show case. It will be linked from documentation.

### Issues Resolved
https://github.com/opensearch-project/opensearch-learning-to-rank-base/issues/43

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by epugh** [comment]
I didn't quite realize that this demo is totally independent of the plugin itself...

File: demo/xgboost-demo/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+This folder contains files demonstrating how to work with xgboost.
+
+## xgboost.txt
+The sample training data utilized in the main demo folder but in LibSVM format which is used by xgboost.
```
Comment (sstults) [inline demo/xgboost-demo/README.md:4]: Are there other demo files? This seems to refer to other files up a directory. If we're only going to include an xgboost demo we should eliminate the extra folder structure.

Comment (epugh) [reply demo/xgboost-demo/README.md:4]: @JohannesDaniel  thoughts on this?    Let's make a decision and get this merged so we can merge the docs!

Comment (JohannesDaniel) [reply demo/xgboost-demo/README.md:4]: This is the source of the files: https://github.com/o19s/elasticsearch-learning-to-rank/tree/main/demo

Comment (JohannesDaniel) [reply demo/xgboost-demo/README.md:4]: xgboost.txt is included: https://github.com/JohannesDaniel/opensearch-learning-to-rank-base/blob/add-ltr-demo/demo/xgboost-demo/xgboost.txt

Comment (JohannesDaniel) [reply demo/xgboost-demo/README.md:4]: I just copied exactly what I found in the link posted above. But I agree with @sstults something might be missing. In the official documentation we have the following link: https://github.com/JohannesDaniel/opensearch-learning-to-rank-base/blob/add-ltr-demo/demo/xgboost-demo/xgboost.txt

But this file is not included here.

Comment (sstults) [reply demo/xgboost-demo/README.md:4]: I think we're fine. It's not that big of a deal.

**Pull request merged**